src/brevitas/graph/gptq.py

```diff
@@ -331,24 +395,33 @@ def single_layer_update(self, percdamp=.01):
             else:
                 perm = permutation_list[0]
                 weight[:, perm[i2:]] -= error_block.matmul(h_inv[0, i1:i2, i2:])
+            weight.to(dtype)
 
     def get_quant_weights(self, i, i1, i2, permutation_list):
         # We need to recompute quant weights at runtime since our float weights are being updated
-        quant_weight = self.layer.quant_weight()
-        quant_weight = quant_weight.value
 
-        if isinstance(self.layer, qnn.QuantConv2d):
-            quant_weight = quant_weight.flatten(1)
+        if self.in_channel_subslice is None:
+            quant_weight = self.layer.quant_weight()
+            quant_weight = quant_weight.value
 
-        if self.act_order:
-            # If act order is enabled, permute quant weight to match the float32 counterpart
-            # len(permutation_list) == self.groups
-            if self.groups == 1:
-                quant_weight = quant_weight[:, permutation_list[0]]
-            else:
-                for ii, perm in enumerate(permutation_list):
-                    quant_weight[ii, :] = quant_weight[ii, perm]
+            if isinstance(self.layer, SUPPORTED_CONV_OP):
+                quant_weight = quant_weight.flatten(1)
+
+            if self.act_order:
+                # If act order is enabled, permute quant weight to match the float32 counterpart
+                # len(permutation_list) == self.groups
+                if self.groups == 1:
+                    quant_weight = quant_weight[:, permutation_list[0]]
+                else:
+                    for ii, perm in enumerate(permutation_list):
+                        quant_weight[ii, :] = quant_weight[ii, perm]
+
+            quant_weight_block = quant_weight[:, i1:i2]
+            q = quant_weight_block[:, i]
+        else:
+            pass
+        # else:
```

Comment:
Can you raise a NotImplemented?

Reply:
This should not be here since it will addressed in another PR entirely, apologies